airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/utils/json_schema_helper.py

```diff
@@ -45,20 +45,27 @@ def get_property(self, path: List[str]):
             node = node["properties"][segment]
         return node
 
-    def get_type_for_key_path(self, path: List[str]):
+    def get_type_for_key_path(self, path: List[str]) -> Set[str]:
+        type_ = []
         try:
-            return self.get_property(path)["type"]
+            field = self.get_property(path)
+            type_ = field.get("format", field["type"])
+            if not isinstance(type_, List):
+                type_ = [type_]
         except KeyError:
-            return None
+            pass
+        return set(type_)
 
     def get_cursor_value(self, record, cursor_path):
         type_ = self.get_type_for_key_path(path=cursor_path)
         value = reduce(lambda data, key: data[key], cursor_path, record)
         return self.parse_value(value, type_)
 
     @staticmethod
-    def parse_value(value, type_):
-        if type_ in ("datetime", "date-time"):
+    def parse_value(value: Any, type_: Set[str]):
+        if type_ & {"datetime", "date-time"}:
+            if value is None and "null" not in type_:
+                raise ValueError("Invalid field format")
```

Comment:
There is a rename comment above but we should make this easier to consume. 

```suggestion
                raise ValueError(f"Invalid field format. Value: {value}. Format: {type_}")
```